README.md

```diff
@@ -16,82 +17,75 @@ And libraries:
+ * `pkg/lib` - providing external interfaces for interacting with this project as an api that defines a set of standards for operator bundles and indexes.
+ * `pkg/containertools` - providing an interface to interact with and shell out to common container tooling binaries (if installed on the environment)
 
 # Manifest format
 
 
-We refer to a directory of files with one ClusterServiceVersion as a "bundle". A bundle is typically just a ClusterServiceVersion and the CRDs that define the owned APIs of the CSV, though additional objects may be included.
+We refer to a directory of files with one ClusterServiceVersion as a "bundle". A bundle typically includes a ClusterServiceVersion and the CRDs that define the owned APIs of the CSV in its manifest directory, though additional objects may be included. It also includes an annotations file in its metadata folder which defines some higher level aggregate data that helps to describe the format and package information about how the bundle should be added into an index of bundles.
 
 ```
  # example bundle
- 0.6.1
- ├── etcdcluster.crd.yaml
- └── etcdoperator.clusterserviceversion.yaml
+ etcd
+ ├── manifests
+ │   ├── etcdcluster.crd.yaml
+ │   └── etcdoperator.clusterserviceversion.yaml
+ └── metadata
+     └── annotations.yaml
 ```
 
 When loading manifests into the database, the following invariants are validated:
 
- * Every package has at least one channel
- * Every ClusterServiceVersion pointed to by a channel in a package exists
+ * The bundle must have at least one channel defined in the annotations.
  * Every bundle has exactly one ClusterServiceVersion.
  * If a ClusterServiceVersion `owns` a CRD, that CRD must exist in the bundle.
- * If a ClusterServiceVersion `replaces` another, both the old and the new must exist in the package.
 
 Bundle directories are identified solely by the fact that they contain a ClusterServiceVersion, which provides an amount of freedom for layout of manifests.
 
-It's recommended to follow a layout that makes it clear which bundles are part of which package, as in [manifests](manifests):
+Check out the [operator bundle design](docs/design/operator-bundle.md) for more detail on the bundle format.
 
+# Bundle images
+
+Using [OCI spec](https://github.com/opencontainers/image-spec/blob/master/spec.md) container images as a method of storing the manifest and metadata contents of individual bundles, `opm` interacts directly with these images to generate and incrementally update the database. Once you have your [manifests defined](https://operator-framework.github.io/olm-book/docs/packaging-an-operator.html#writing-your-operator-manifests) and have created a directory in the format defined above, building the image is as simple as defining a [Dockerfile](docs/design/operator-bundle.md#Bundle-Dockerfile) and building that image:
```

Comment:
I'm unclear on how flexible the directory structure is intended to be. Is a distinction made between a bundle directory structure described above and the layout of files in a bundle image?

I see the statement "bundle directories are identified solely by the fact that they contain a ClusterServiceVersion, which provides an amount of freedom for layout of manifests," and I'm not sure how that aligns with the proscribed metadata and manifests directories mentioned above.

Reply:
The only directory we care about is the one on the image. You could, in theory, write your own Dockerfile that pulls files from lots of places to generate the directory structure in place. The flexibility is that the manifest directory can have certain other kube manifests optionally (currently that list includes rbac files and some prometheus stuff aside from the CSVs and CRDs). In future iterations this list can (and will be) extended. But all of that does need to live in the `manifests/` subdirectory on your image.

Right now `metadata/` has just the `annotations.yaml` file, but we are intending for this to be extended in the same fashion.